core/state/statedb.go

```diff
@@ -592,6 +597,11 @@ func (db *DB) GetRefund() uint64 {
 // Finalise finalises the state by removing the db destructed objects
 // and clears the journal as well as the refunds.
 func (db *DB) Finalise(deleteEmptyObjects bool) {
```

Comment:
what should be the behavior for graceful shutdown? flush the cache or inconsistent state is okay?

Reply:
The in-memory data in a state being processed is ephemeral data before the state is finalized and confirmed. So no need to worry about graceful shutdown (same for the existing stateObjects logic)